doc/source/timeseries.rst

```diff
@@ -110,12 +110,13 @@ However, :class:`Series` and :class:`DataFrame` can directly also support the ti
 
    pd.Series(pd.date_range('2000', freq='D', periods=3))
 
-:class:`Series` and :class:`DataFrame` have extended data type support and functionality for ``datetime`` and ``timedelta``
-data when the time data is used as data itself. The ``Period`` and ``DateOffset`` data will be stored as ``object`` data.
+:class:`Series` and :class:`DataFrame` have extended data type support and functionality for ``datetime``, ``timedelta``
+and ``Period`` data when passed into those constructors. ``DateOffset``
```

Comment:
May need to be careful about this, as Period only supports a subset of DateOffsets, which has caused confusion in the past.

Reply:
Good to know. This section is more describing what dtypes are supported by Series and DataFrames (i.e. Period dtype is supported but not DateOffset (since it doesnt really have its own data type))